tests/vmi_multus_test.go

```diff
@@ -885,6 +872,56 @@ var _ = Describe("[Serial]SRIOV", func() {
 				return libnet.PingFromVMConsole(vmi2, cidrToIP(cidrA))
 			}, 15*time.Second, time.Second).Should(Succeed())
 		})
+
+		Context("With VLAN", func() {
+			var (
+				macVlaned1   string
+				macVlaned2   string
+				macNonVlaned string
+			)
+
+			const (
+				cidrVlaned1          = "192.168.0.1/24"
+				cidrNonVlaned        = "192.168.0.3/24"
+				cidrVlaned2          = "192.168.0.2/24"
+				sriovVlanNetworkName = "sriov-vlan"
+			)
+
+			generateRandomMacAsString := func() string {
+				mac, err := tests.GenerateRandomMac()
+				Expect(err).ToNot(HaveOccurred())
+				return mac.String()
+			}
+
+			BeforeEach(func() {
+				createNetworkAttachementDefinition(sriovVlanNetworkName, tests.NamespaceTestDefault, sriovConfVlanCRD)
+				macVlaned1 = generateRandomMacAsString()
+				macVlaned2 = generateRandomMacAsString()
+				macNonVlaned = generateRandomMacAsString()
+			})
+
+			It("should be able to ping between two VMIs with the same VLAN over SRIOV network", func() {
+				// create an anonymous vmi to be used for pinging
+				createSriovVMIIWithStaticIPByMac(sriovVlanNetworkName, macVlaned1, cidrVlaned1)
+				vlanedVMI := createSriovVMIIWithStaticIPByMac(sriovVlanNetworkName, macVlaned2, cidrVlaned2)
+
+				By("pinging between vlanedVMI1 and vlanedVMI2 over vlan")
+				Eventually(func() error {
+					return libnet.PingFromVMConsole(vlanedVMI, cidrToIP(cidrVlaned1))
```

Comment:
Why the ping command is inside an `Eventually`?

Reply:
https://github.com/kubevirt/kubevirt/pull/4606 it is all explained here. I adopted @qinqon  approach.

Reply:
I guess we will stick with this approach until we could relay on the VMI.status for SRIOV interfaces

Reply:
OK. Make sense . But I think it worth a separate PR.

Reply:
`PingFromVMConsole` is varidic function it enable you to pass arguments to `ping` command,
so you could pass `-c 4 -w 7` to  PingFromVMConsole so it will send more then one packet
https://github.com/kubevirt/kubevirt/blob/master/tests/libnet/ping.go#L37